In [2]:
import pandas as pd
import seaborn as sns

In [3]:
df = pd.read_csv('student_data.csv')

In [4]:
df.head(5)

Marital status  Application mode  Application order  Course  \
0              1                17                  5   171.0   
1              1                15                  1  9254.0   
2              1                 1                  5  9070.0   
3              1                17                  2  9773.0   
4              2                39                  1  8014.0   

   Daytime/evening attendance  Previous qualification  \
0                           1                       1   
1                           1                       1   
2                           1                       1   
3                           1                       1   
4                           0                       1   

   Previous qualification (grade)  Nacionality  Mother's qualification  \
0                           122.0            1                      19   
1                           160.0            1                       1   
2                           122.0            1                      37   
3                           122.0            1                      38   
4                           100.0            1                      37   

   Father's qualification  ...  Curricular units 2nd sem (credited)  \
0                    12.0  ...                                    0   
1                     3.0  ...                                    0   
2                    37.0  ...                                    0   
3                    37.0  ...                                    0   
4                    38.0  ...                                    0   

   Curricular units 2nd sem (enrolled)  \
0                                  0.0   
1                                  6.0   
2                                  6.0   
3                                  6.0   
4                                  6.0   

   Curricular units 2nd sem (evaluations) Curricular units 2nd sem (approved)  \
0                                       0                                   0   
1                                       6                                   6   
2                                       0                                   0   
3                                      10                                   5   
4                                       6                                   6   

   Curricular units 2nd sem (grade)  \
0                          0.000000   
1                         13.666667   
2                          0.000000   
3                         12.400000   
4                         13.000000   

   Curricular units 2nd sem (without evaluations) Unemployment rate  \
0                                               0              10.8   
1                                               0              13.9   
2                                               0              10.8   
3                                               0               9.4   
4                                               0              13.9   

   Inflation rate   GDP    Target  
0             1.4  1.74   Dropout  
1            -0.3  0.79  Graduate  
2             1.4  1.74   Dropout  
3            -0.8 -3.12  Graduate  
4            -0.3  0.79  Graduate  

[5 rows x 37 columns]

The first five rows of the dataset show the structure and the format of the data. This data contains a mixture of categorical and continuous features, such as Marital status, Application Mode, Application Order and course, as columns. 

The continuous features are Previous qualification (grade), Admission grade, Unemployment rate, Inflation rate and GDP. 

The categorical features are the majority of them. These include demographic information, such as Merital Status, Nacionality, Mother's qualification; course information, such as Application Mode, Applications order, Course and Daytime/evening attendance; and other categorical features.   

In [5]:
df.shape

(4424, 37)

The dataset contains 4424 obeservations (rows) and 37 attributes (columns). 

In [6]:
df.describe()

Application mode  Application order       Course  \
count       4424.000000        4424.000000  4423.000000   
mean          18.669078           1.727848  8856.458738   
std           17.484682           1.313793  2063.763479   
min            1.000000           0.000000    33.000000   
25%            1.000000           1.000000  9085.000000   
50%           17.000000           1.000000  9238.000000   
75%           39.000000           2.000000  9556.000000   
max           57.000000           9.000000  9991.000000   

       Daytime/evening attendance  Previous qualification  \
count                 4424.000000             4424.000000   
mean                     0.890823                4.577758   
std                      0.311897               10.216592   
min                      0.000000                1.000000   
25%                      1.000000                1.000000   
50%                      1.000000                1.000000   
75%                      1.000000                1.000000   
max                      1.000000               43.000000   

       Previous qualification (grade)  Nacionality  Mother's qualification  \
count                     4424.000000  4424.000000             4424.000000   
mean                       132.613314     1.873192               19.561935   
std                         13.188332     6.914514               15.603186   
min                         95.000000     1.000000                1.000000   
25%                        125.000000     1.000000                2.000000   
50%                        133.100000     1.000000               19.000000   
75%                        140.000000     1.000000               37.000000   
max                        190.000000   109.000000               44.000000   

       Father's qualification  Mother's occupation  ...  \
count             4423.000000          4424.000000  ...   
mean                22.271987            10.960895  ...   
std                 15.343244            26.418253  ...   
min                  1.000000             0.000000  ...   
25%                  3.000000             4.000000  ...   
50%                 19.000000             5.000000  ...   
75%                 37.000000             9.000000  ...   
max                 44.000000           194.000000  ...   

       Curricular units 1st sem (without evaluations)  \
count                                     4424.000000   
mean                                         0.137658   
std                                          0.690880   
min                                          0.000000   
25%                                          0.000000   
50%                                          0.000000   
75%                                          0.000000   
max                                         12.000000   

       Curricular units 2nd sem (credited)  \
count                          4424.000000   
mean                              0.541817   
std                               1.918546   
min                               0.000000   
25%                               0.000000   
50%                               0.000000   
75%                               0.000000   
max                              19.000000   

       Curricular units 2nd sem (enrolled)  \
count                          4423.000000   
mean                              6.231743   
std                               2.196038   
min                               0.000000   
25%                               5.000000   
50%                               6.000000   
75%                               7.000000   
max                              23.000000   

       Curricular units 2nd sem (evaluations)  \
count                             4424.000000   
mean                                 8.063291   
std                                  3.947951   
min                                  0.000000   
25%                                  6.000000   
50%                                  8.000000   
75%

The descriptive statistics reveal important patterns in numerical features. 

By compering the mean and the median, we can assess weather the data is symmetrically distributed or contains outliers. When the median and mediun are similar, this means the data is normally distributed. However, when differ significantly, it suggest the presence of outilers.

In [11]:
df.describe(include=object)


Marital status Displaced Tuition fees up to date Age at enrollment  \
count            4424      4424                    4423              4424   
unique              7         3                       3                47   
top                 1         1                       1                18   
freq             3918      2425                    3895              1036   

       Curricular units 1st sem (credited)    Target  
count                                 4424      4424  
unique                                  22         7  
top                                      0  Graduate  
freq                                  3847      2207

This output reveals data about columns stored as text.

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4424 entries, 0 to 4423
Data columns (total 37 columns):
 #   Column                                          Non-Null Count  Dtype  
---  ------                                          --------------  -----  
 0   Marital status                                  4424 non-null   object 
 1   Application mode                                4424 non-null   int64  
 2   Application order                               4424 non-null   int64  
 3   Course                                          4423 non-null   float64
 4   Daytime/evening attendance                      4424 non-null   int64  
 5   Previous qualification                          4424 non-null   int64  
 6   Previous qualification (grade)                  4424 non-null   float64
 7   Nacionality                                     4424 non-null   int64  
 8   Mother's qualification                          4424 non-null   int64  
 9   Father's qualification                   

In [9]:
df.isnull().sum()

Marital status                                    0
Application mode                                  0
Application order                                 0
Course                                            1
Daytime/evening attendance                        0
Previous qualification                            0
Previous qualification (grade)                    0
Nacionality                                       0
Mother's qualification                            0
Father's qualification                            1
Mother's occupation                               0
Father's occupation                               0
Admission grade                                   1
Displaced                                         0
Educational special needs                         0
Debtor                                            1
Tuition fees up to date                           1
Gender                                            0
Scholarship holder                                0
Age at enrol